File: frontend/types/referral-feedback.ts
export type FeedbackStatus = "PENDING" | "ACCEPTED" | "REJECTED" | "COMPLETED";
export type TimelineStatusType = "WORKFLOW" | "FEEDBACK";

export interface ReferralTimelineEntry {
  id: number;
  referral_id: number;
  actor_id: number;
  from_status?: string;
  to_status: string;
  status_type: TimelineStatusType;
  note?: string;
  created_at: string;
  actor?: {
    id: number;
    name: string;
    email: string;
    role: string;
  };
}

export interface ReferralFeedback {
  referral_id: number;
  feedback_status: FeedbackStatus;
  accepted_at?: string;
  completed_at?: string;
  outcome_summary?: string;
  recommendation_text?: string;
  referring_physician_id?: number;
  timeline: ReferralTimelineEntry[];
}

export const FEEDBACK_STATUS_LABELS: Record<FeedbackStatus, string> = {
  PENDING: "Awaiting Specialist",
  ACCEPTED: "Accepted",
  REJECTED: "Rejected",
  COMPLETED: "Completed",
};

export const FEEDBACK_STATUS_COLORS: Record<FeedbackStatus, string> = {
  PENDING: "bg-yellow-100 text-yellow-800 border-yellow-200",
  ACCEPTED: "bg-blue-100 text-blue-800 border-blue-200",
  REJECTED: "bg-red-100 text-red-800 border-red-200",
  COMPLETED: "bg-green-100 text-green-800 border-green-200",
};

File: backend/app/schemas/referral_feedback.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from app.models.referral import FeedbackStatus
from app.models.referral_timeline import TimelineStatusType
from app.schemas import UserOut


class ReferralTimelineOut(BaseModel):
    id: int
    referral_id: int
    actor_id: int
    from_status: Optional[str] = None
    to_status: str
    status_type: TimelineStatusType
    note: Optional[str] = None
    created_at: datetime
    actor: Optional[UserOut] = None

    class Config:
        from_attributes = True


class ReferralFeedbackAccept(BaseModel):
    note: Optional[str] = None


class ReferralFeedbackReject(BaseModel):
    note: str = Field(..., min_length=3, max_length=2000)


class ReferralFeedbackComplete(BaseModel):
    outcome_summary: str = Field(..., min_length=3, max_length=5000)
    recommendation_text: str = Field(..., min_length=3, max_length=5000)


class ReferralFeedbackOut(BaseModel):
    referral_id: int
    feedback_status: FeedbackStatus
    accepted_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    outcome_summary: Optional[str] = None
    recommendation_text: Optional[str] = None
    referring_physician_id: Optional[int] = None
    timeline: List[ReferralTimelineOut] = []
